runtime/interp.c

```diff
@@ -81,7 +81,7 @@ sp is a local copy of the global variable Caml_state->extern_sp. */
   { sp = domain_state->current_stack->sp; accu = sp[0]; env = sp[1]; sp += 3; }
 #define Enter_gc \
   { Setup_for_gc; caml_handle_gc_interrupt(); \
-    caml_check_for_pending_signals();         \
+    caml_process_pending_actions();         \
```

Comment:
`caml_process_pending_actions` already has a call for to `caml_handle_gc_interrupt` (actions is just gc_interrupt+pending_signals), so probably worth reverting this change or removing the `caml_handle_gc_interrupt`.